firecares/tasks/update.py

```diff
@@ -448,3 +453,124 @@ def expand(values, weights):
+    """
+    Task for updating the servicearea table rolling up parcel hazard categories with departement drive time data
+    """
+    for fd in FireDepartment.objects.all():
+        get_parcel_department_hazard_level_rollup(fd.id)
+
+
+def get_parcel_department_hazard_level_rollup(fd_id=100214):
+    """
+    Update for one department for the drive time hazard level
+    """
+    stationlist = FireStation.objects.filter(department_id=fd_id)
+    dept = FireDepartment.objects.filter(id=fd_id)
+
+    print "Calculating Drive times for:  " + dept[0].name
+
+    #  Use Headquarters geometry if there is no Statffing assets
+    if len(stationlist) < 1:
+        drivetimeurl = 'https://geo.firecares.org/?f=json&Facilities={"features":[{"geometry":{"x":' + str(dept[0].headquarters_geom.x) + ',"spatialReference":{"wkid":4326},"y":' + str(dept[0].headquarters_geom.y) + '}}],"geometryType":"esriGeometryPoint"}&env:outSR=4326&text_input=4&Break_Values=4 6 8&returnZ=false&returnM=false'
+
+    else:
+        drivetimegeom = {}
+        for fireStation in stationlist:
+            stationasset = {}
+            stationasset["spatialReference"] = {"wkid": 4326}
+            stationasset["y"] = round(fireStation.geom.y, 5)
+            stationasset["x"] = round(fireStation.geom.x, 5)
+            stationgeom = {}
+            stationgeom["geometry"] = stationasset
+            drivetimegeom.update(stationgeom)
+
+        drivetimeurl = 'https://geo.firecares.org/?f=json&Facilities={"features":[' + json.dumps(drivetimegeom) + '],"geometryType":"esriGeometryPoint"}&env:outSR=4326&text_input=4&Break_Values=4 6 8&returnZ=false&returnM=false'
+
+    try:
+        getdrivetime = requests.get(drivetimeurl)
+        update_parcel_department_hazard_level(json.loads(getdrivetime.content)['results'][0]['value']['features'], dept[0])
+
+    except KeyError:
+        print 'Drive Time Failed for ' + dept[0].name
+        print drivetimeurl
+
+    except IntegrityError:
+        print 'Drive Time Failed for ' + dept[0].name
+        print drivetimeurl
+
+    except:
+        print 'Drive Time Failed for ' + dept[0].name
+        print drivetimeurl
+
+
+def update_parcel_department_hazard_level(drivetimegeom, department):
+    """
+    Intersect with Parcel layer and update parcel_department_hazard_level table
+    0-4 minutes
+    4-6 minutes
+    6-8 minutes
+    """
+
+    drivetimegeom0 = arcgis2geojson(drivetimegeom[2]['geometry'])
+    drivetimegeom4 = arcgis2geojson(drivetimegeom[1]['geometry'])
+    drivetimegeom6 = arcgis2geojson(drivetimegeom[0]['geometry'])
+
+    cursor = connections['nfirs'].cursor()
+
+    QUERY_INTERSECT_FOR_PARCEL_DRIVETIME = """SELECT sum(case when l.risk_category = 'Low' THEN 1 ELSE 0 END) as low,
+        sum(CASE WHEN l.risk_category = 'Medium' THEN 1 ELSE 0 END) as medium,
+        sum(CASE WHEN l.risk_category = 'High' THEN 1 ELSE 0 END) high,
+        sum(CASE WHEN l.risk_category is null THEN 1 ELSE 0 END) as unknown
```

Comment:
I doubt this is the case, but you might want to check that there aren't any parcels that have a `risk_category = trim('')`, if _ALL_ are `null` then we're good.

Reply:
select distinct risk_category, count(risk_category) from parcel_risk_category_local group by risk_category  

no value other than null, low, medium and high